prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -245,8 +252,7 @@ def search_and_save_roles(
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
-            # TODO: add custom exception once we have the Kubernetes exceptions
-            raise error
+            raise KubernetesSearchAndSaveRolesError(original_exception=error)
```

Comment:
I'm not sure about this kind of exception, since in this function we are just getting and setting values in a list.

Reply:
I think so, in this case there is nothing specifically related to K8s happening in the function.